java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -397,9 +411,13 @@ public void visitThrowStatement(ThrowStatementTree tree) {
 
   @Override
   public void visitLambdaExpression(LambdaExpressionTree tree) {
-    //TODO resolve variables
-    super.visitLambdaExpression(tree);
-    registerType(tree, Symbols.unknownType);
+    LambdaExpressionTreeImpl lambdaExpressionTree = (LambdaExpressionTreeImpl) tree;
+    if(lambdaExpressionTree.isTypeSet()) {
+      //FIXME Based on set type get correct type of lambda var if required.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)